### PR TITLE
added dice_loss #18231

### PR DESCRIPTION
**Pull request by zeus2x7**
added dice loss in paddle.nn.functional.loss referring to issue https://github.com/unifyai/ivy/issues/18231
close https://github.com/unifyai/ivy/issues/18231
sub-task https://github.com/unifyai/ivy/issues/15053

### CODE REVIEW

**Comment by zeus2x7** [comment]
hi @OmarJabri7  can you please look into the pr  ?
